sonarts-core/src/cfg/cfg.ts

```diff
@@ -122,17 +128,29 @@ export interface CfgBlock {
   replacePredecessor(what: CfgBlock, withWhat: CfgBlock): void;
 
   getLabel(): string;
+
+  getId(): number;
 }
 
 export abstract class CfgBlockWithPredecessors {
   public predecessors: CfgBlock[] = [];
   public loopingStatement: ts.IterationStatement | undefined;
   public branchingElement: ts.Node | undefined = undefined;
+  public successorWithoutJump: CfgBlock | undefined;
+  private static idGen = 0;
+  private id: number | undefined = undefined;
 
   public replacePredecessor(what: CfgBlock, withWhat: CfgBlock): void {
     const index = this.predecessors.indexOf(what);
     this.predecessors[index] = withWhat;
   }
+
+  public getId(): number {
+    if (this.id == undefined) {
+      this.id = CfgBlockWithPredecessors.idGen++;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this increment operation into a dedicated statement. ([typescript:S881](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS881))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=628&issues=AWNE6bCtlRUrfRWxk6O3&open=AWNE6bCtlRUrfRWxk6O3)

[ISSUE_KEY]: AWNE6bCtlRUrfRWxk6O3